hawc/apps/eco/views.py

```diff
@@ -0,0 +1,173 @@
+from django.http import HttpRequest
+from django.shortcuts import render
+
+from ..common.htmx import HtmxViewSet, action, can_edit, can_view
+from ..common.views import BaseCreate, BaseDelete, BaseDetail, BaseUpdate
+from ..study.models import Study
+from . import forms, models
+
+
+# Design
+class DesignCreate(BaseCreate):
+    success_message = "Study design created."
+    parent_model = Study
+    parent_template_name = "study"
+    model = models.Design
+    form_class = forms.DesignForm
+
+    def get_success_url(self):
+        return self.object.get_update_url()
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        return context
+
+
+class DesignUpdate(BaseUpdate):
+    success_message = "Study design updated."
+    parent_model = Study
+    parent_template_name = "study"
+    model = models.Design
+    form_class = forms.DesignForm
+    template_name = "eco/design_update.html"
+
+    def get_context_data(self, **kwargs):
+        design_id = self.kwargs.pop("pk")
+        context = super().get_context_data(**kwargs)
+        study_id = self.object.study_id
+        context["causes"] = models.Cause.objects.filter(study=study_id)
```

Comment:
We can remove these 3 context items for update, detail, and delete. Instead, update the `get_queryset` method and just prefetch_related for these models; and then we can use the attributes directly on the instance